packages/@lwc/state/src/__tests__/index.test.ts

```diff
@@ -98,4 +132,67 @@ describe('state manager', () => {
       s.value.count = 10;
     }).toThrow();
   });
+
+  test('should call computeds notify only after an update and not at init', () => {
+    const s = state(1, 'Apple');
+    expect(doubleCountNotifySpy).not.toHaveBeenCalled();
+    expect(fruitNameAndCountNotifySpy).not.toHaveBeenCalled();
+
+    s.value.increment();
+    expect(doubleCountNotifySpy).toHaveBeenCalledOnce();
+    expect(fruitNameAndCountNotifySpy).toHaveBeenCalledOnce();
+  });
+
+  test('should only notify computeds that depends on the atom being updated', () => {
+    const s = state(5, 'Apple');
+
+    s.value.increment();
+    expect(doubleCountNotifySpy).toHaveBeenCalledOnce();
+    expect(fruitNameAndCountNotifySpy).toHaveBeenCalledOnce();
+
+    s.value.changeFruit('Banana');
+    // should not be notified
+    expect(doubleCountNotifySpy).toHaveBeenCalledOnce();
+    expect(fruitNameAndCountNotifySpy).toHaveBeenCalledTimes(2);
+  });
+
+  test('should compute computeds value lazily', () => {
+    const s = state(5, 'Apple');
+    expect(fruitNameAndCountComputeValueSpy).not.toHaveBeenCalled();
+
+    // every time we access state managers value
+    // all stale values are re-computed
+    s.value.increment();
+    expect(fruitNameAndCountComputeValueSpy).toHaveBeenCalledOnce();
+    s.value.changeFruit('Banana');
+    expect(fruitNameAndCountComputeValueSpy).toHaveBeenCalledTimes(2);
+  });
+
+  test('should not compute value until value is not accessed through getter', () => {
+    const s = state(5, 'Apple');
+    // getting references to the updaters and computed
+    const { increment, changeFruit, fruitNameAndCount } = s.value;
+
+    expect(fruitNameAndCountComputeValueSpy).toHaveBeenCalledOnce();
+
+    increment();
+    changeFruit('Banana');
+    // not re-computed since value is not accessed
```

Comment:
@divmain I suppose this is the limitation that we will have due to lazy-compute since we aren't accessing the state managers getter and the value isn't updated until we access the getter.

Reply:
Yep, this is exactly what we want actually. If nobody is trying to get at the data, we should not waste CPU cycles recomputing things. If someone had subscribed to the state manager's signal, it would be recomputed. It might be worth adding a test to confirm.